tests/test_cli.py

```diff
@@ -263,6 +318,82 @@ def test_cli_start_with_input(runner):
+                              'mcr.microsoft.com/azureiotedge-hub:1.0',
+                              'hello-world'])
+
+
+@pytest.mark.skipif(get_docker_os_type() == 'windows', reason='The base image of edgeHubDev image is 1809 but agent is 1803.'
+                    'So it does not support windows container.')
+def test_cli_start_with_registry(runner):
+    print("Please provision a regsitry with amd64 module and windows-amd64 module before run this test case.")
+    try:
+        if get_docker_os_type() == "windows":
+            deployment_json_file_path = os.path.join(test_config_dir, "deployment.windows-amd64.json")
+            module_image_name = ('"image": "{0}/{1}"').format(VALID_CONTAINERREGISTRYSERVER,
+                                                              'windows_filtermodule_with_registry:0.0.1-windows-amd64')
+            update_file_content(deployment_json_file_path, '"image": ""', module_image_name)
+        else:
+            deployment_json_file_path = os.path.join(test_config_dir, "deployment.amd64.json")
+            module_image_name = ('"image": "{0}/{1}"').format(VALID_CONTAINERREGISTRYSERVER,
+                                                              'filtermodule_with_registry:0.0.1-amd64')
+            update_file_content(deployment_json_file_path, '"image": ""', module_image_name)
+
+        update_file_content(deployment_json_file_path, '"username": ""',
+                            '"username": "' + VALID_CONTAINERREGISTRYUSERNAME + '"')
+        update_file_content(deployment_json_file_path, '"password": ""',
+                            '"password": "' + VALID_CONTAINERREGISTRYPASSWORD + '"')
+        update_file_content(deployment_json_file_path, '"address": ""',
+                            '"address": "' + VALID_CONTAINERREGISTRYSERVER + '"')
+
+        cli_start_with_deployment(runner, deployment_json_file_path)
+        if get_docker_os_type() == "windows":
+            wait_verify_docker_output(['docker', 'ps'], ['windows_filtermodule_with_registry'])
+        else:
+            wait_verify_docker_output(['docker', 'ps'], ['filtermodule_with_registry'])
+    finally:
+        update_file_content(deployment_json_file_path, module_image_name, '"image": ""')
+        update_file_content(deployment_json_file_path, '"username": "(.*)"', '"username": ""')
+        update_file_content(deployment_json_file_path, '"password": "(.*)"', '"password": ""')
+        update_file_content(deployment_json_file_path, '"address": "(.*)"', '"address": ""')
+
+        result = cli_stop(runner)
+        assert 'IoT Edge Simulator has been stopped successfully' in result.output.strip()
+
+        remove_docker_networks(['azure-iot-edge-dev'])
+        remove_docker_images(['mcr.microsoft.com/azureiotedge-simulated-temperature-sensor:1.0',
+                              'mcr.microsoft.com/azureiotedge-hub:1.0',
+                              'aziotedgetoolingtest.azurecr.io/filtermodule_with_registry:0.0.1-amd64',
```

Comment:
the image is from repo aziotedgetoolingtest.azurecr.io? or it should be the VALID_CONTAINERREGISTRYSERVER?

Reply:
Updated code